src/main/java/com/Elessar/database/MongoDB.java

```diff
@@ -0,0 +1,111 @@
+package com.Elessar.database;
+
+import com.Elessar.app.server.User;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import java.util.ArrayList;
+import java.util.List;
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+
+/**
+ * Created by Hans on 1/20/19.
+ */
+public class MongoDB implements MyDatabase{
+    private final MongoDatabase db;
+    public MongoDB(MongoDatabase db) {
+        this.db = db;
+    }
+
+    @Override
+    public void insert(User user) {
+        final Document doc = new Document();
+        doc.append("name", user.getName());
+        doc.append("password", user.getPassword());
+        doc.append("email", user.getEmail());
+        doc.append("online", user.getOnline());
+        if (user.getPhoneNumber().length() > 0) {
+            doc.append("phone", user.getPhoneNumber());
+        }
+        db.getCollection("users").insertOne(doc);
+    }
+
+    @Override
+    public List<User> findUsers(User filter) {
+        List<User> findResult = new ArrayList<>();
+        try (MongoCursor<Document> cursor = db.getCollection("users").find(userToBson(filter)).iterator()) {
+            while (cursor.hasNext()) {
+                findResult.add(docToUser(cursor.next()));
+            }
+        }
+        return findResult;
+
+    }
+
+    @Override
+    public User update(User user) {
+        Bson filter = eq("name", user.getName());
+        if (user.getPassword() != null) {
+            filter = and(filter, eq("password", user.getPassword()));
+        }
+        Document prevUserDoc = db.getCollection("users").findOneAndUpdate(filter, new Document("$set", userToDoc(user)));
```

Comment:
Since we search the DB based on user name (and password), even we leave user name there, does it hurt? The database will just replace user name on name field with the same value.

Reply:
Can I just change user.name = null, or I need to recreate a new user instance before calling userToDoc(user)?